src/groupeddataframe/grouping.jl

```diff
@@ -1214,3 +1278,47 @@ groupindices(gd::GroupedDataFrame) = replace(gd.groups, 0=>missing)
 Return a vector of column names in `parent(gd)` used for grouping.
 """
 groupvars(gd::GroupedDataFrame) = _names(gd)[gd.cols]
+
+"""
+    groupvalues(gd::GroupedDataFrame)
+
+Get the values of the grouped columns for each group.
+
+### Returns
+
+A [`DataFrame`](@ref) with the grouped columns and rows corresponding to groups.
+
+### Examples
+
+```jldoctest
+julia> df = DataFrame(a = repeat([:foo, :bar, :baz], outer=[2]),
+                      b = repeat([2, 1], outer=[3]),
+                      c = 1:6);
+
+julia> gd = groupby(df, :a)
+GroupedDataFrame with 3 groups based on key: a
+First Group (2 rows): a = :foo
+│ Row │ a      │ b     │ c     │
+│     │ Symbol │ Int64 │ Int64 │
+├─────┼────────┼───────┼───────┤
+│ 1   │ foo    │ 2     │ 1     │
+│ 2   │ foo    │ 1     │ 4     │
+⋮
+Last Group (2 rows): a = :baz
+│ Row │ a      │ b     │ c     │
+│     │ Symbol │ Int64 │ Int64 │
+├─────┼────────┼───────┼───────┤
+│ 1   │ baz    │ 2     │ 3     │
+│ 2   │ baz    │ 1     │ 6     │
+
+julia> groupvalues(gd)
+3×1 DataFrame
+│ Row │ a      │
+│     │ Symbol │
+├─────┼────────┤
+│ 1   │ foo    │
+│ 2   │ bar    │
+│ 3   │ baz    │
+```
+"""
+groupvalues(gd::GroupedDataFrame) = gd.values
```

Comment:
If we return here a `DataFrame` (no matter what we store in `values` and if we store anything) then above in indexing we should allow `DataFrameRow` in `getindex` also (as it is a mutable dual of a `NamedTuple`).

But I am not sure if we need this function at all, as it serves the same purpose as `keys` (and `keys` returns `NamedTuples`, so then we do not need to support `DataFrameRow` in `getinex`).

Reply:
Maybe just store the values as a tuple of column vectors? That seems to be the most direct way to create a DataFrame, and it can get the names from the parent each time it's called.